### PR TITLE
Update `fluid-framework` to bundle re-exported modules in its report

### PR DESCRIPTION
**Pull request by Josmithr**
See the updated `.api.md` file for an example of the changed generated report. This model will make it substantially easier to detect breaking changes affecting our public package.

This also cleans up the generated documentation for the website. If you wish to see the complete change, you can build the site locally. But a quick sample of the new output looks like this:
image

Making the API docs for the package appear like those for our other package, and making the fact that the package is implemented by re-exporting other packages an implementation detail, rather than something user-facing.

This PR also updates the `@packageDocumentation` comment for the package to enumerate more:

```typescript
/**
 * The `fluid-framework` package is the primary entry-point into the Fluid client ecosystem when paired with
 * a corresponding service client package (for example, `@fluidframework/azure-client` or
 * `@fluidframework/tinylicious-client`).
 *
 * It contains necessary types and functionality for creating and loading
 * {@link https://fluidframework.com/docs/build/containers/ | Containers}.
 *
 * It also includes a few {@link https://fluidframework.com/docs/build/dds/ | Distributed Data Structures (DDSes)}
 * to get you started using Fluid. These include:
 *
 * - {@link @fluidframework/map#SharedMap}
 *
 * - {@link @fluidframework/map#SharedDirectory}
 *
 * - {@link @fluidframework/sequence#SharedString}
 *
 * Other DDSes and related libraries are published separately.
 *
 * @remarks This package is implemented by re-exporting contents from the following library packages:
 *
 * - `@fluidframwork/container-definitions`
 *
 * - `@fluidframwork/container-loader`
 *
 * - `@fluidframwork/fluid-static`
 *
 * - `@fluidframwork/map`
 *
 * - `@fluidframwork/sequence`
 *
 * @packageDocumentation
 */
```

Other changes:
- Add API-Extractor ("build:docs") to standard build script to ensure reports are re-built alongside changes.
- Fix the "clean" script, which was not deleting some build output
- Move package re-exports directly into `index.ts` 
- Update links in docs to fix new title-wise ambiguity

### CODE REVIEW

File: docs/_includes/links.md [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 <!-- Classes and interfaces -->
 
-[FluidContainer]: {{< relref "fluidcontainer.md" >}}
-[IFluidContainer]: {{< relref "ifluidcontainer.md" >}}
+[FluidContainer]: {{< relref "/docs/apis/fluid-framework/fluidcontainer.md" >}}
+[IFluidContainer]: {{< relref "/docs/apis/fluid-framework/ifluidcontainer.md" >}}
```
Comment (tylerbutler) [inline docs/_includes/links.md:10]: Is there some reason the full path is needed now?

File: docs/content/docs/updates/pre-v1.0.0.md [hunk 2]
```diff
@@ -239,6 +239,6 @@ better reflect their experimental state. If you use these classes, you can add a
 ### Other notable changes
 
 - The published {{< apiref "fluid-framework" >}} package now publishes ESNext modules ({{< issue 7474 >}}).
-- Various APIs in the {{< apiref "azure-client" >}} and {{< apiref "tinylicious-client" >}} now return {{< apiref
-  "IFluidContainer" >}}s instead of {{< apiref "FluidContainer" >}}s. This change should have no effect to developers
-  since the runtime object is the same ({{< issue 7457 >}}).
+- Various APIs in the {{< apiref "azure-client" >}} and {{< apiref "tinylicious-client" >}} now return {{< relref
+  "/docs/apis/fluid-framework/ifluidcontainer.md" >}}s instead of {{< relref "/docs/apis/fluid-framework/fluidcontainer.md" >}}s.
+  This change should have no effect to developers since the runtime object is the same ({{< issue 7457 >}}).
```
Comment (tylerbutler) [inline docs/content/docs/updates/pre-v1.0.0.md:244]: Ideally the api-ref shortcode would continue to work.

Comment (tylerbutler) [reply docs/content/docs/updates/pre-v1.0.0.md:244]: Actually I think this will just output a bare URL which probably isn't what you want.

File: packages/framework/fluid-framework/src/index.ts [hunk 3]
```diff
@@ -4,15 +4,29 @@
...
-export * from "./containerDefinitions";
-export * from "./containerLoader";
-export * from "./fluidStatic";
-export * from "./map";
-export * from "./sequence";
+ export { AttachState } from "@fluidframework/container-definitions";
+export { ConnectionState } from "@fluidframework/container-loader";
+export * from "@fluidframework/fluid-static";
+export * from "@fluidframework/map";
+export * from "@fluidframework/sequence";
```
Comment (tylerbutler) [inline packages/framework/fluid-framework/src/index.ts:32]: Was this change needed?

**Review by tylerbutler — APPROVED** [review]
How will this affect packages that refer to the re-exported libraries directly? E.g. if we ever publish the container-definitions package, and other packages refer to AttachState directly in it vs through the fluid-framework package, will the links go to the right place?

File: docs/content/docs/updates/pre-v1.0.0.md (hunk 2 above)
Comment (Josmithr) [reply docs/content/docs/updates/pre-v1.0.0.md:244]: Where `apiRef` defined? A brief search didn't reveal anything. Maybe its definition just needs to be updated?

File: packages/framework/fluid-framework/api-extractor.json [hunk 4]
```diff
@@ -1,4 +1,11 @@
 {
   "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
-  "extends": "@fluidframework/build-common/api-extractor-common-strict.json"
+  "extends": "@fluidframework/build-common/api-extractor-common-report.json",
```
Comment (Josmithr) [inline packages/framework/fluid-framework/api-extractor.json:3]: Note: this change is required until all the packages we are bundling here also conform to the strict config.

File: packages/framework/fluid-framework/src/index.ts (hunk 3 above)
Comment (Josmithr) [reply packages/framework/fluid-framework/src/index.ts:32]: Not strictly, but I think it makes reasoning about what is in the package easier. I can revert if we like the old structure better, but IMO it just obfuscates things.

File: packages/framework/fluid-framework/src/index.ts [hunk 5]
```diff
@@ -4,15 +4,29 @@
...
  *
  * @packageDocumentation
  */
 
-export * from "./containerDefinitions";
-export * from "./containerLoader";
-export * from "./fluidStatic";
-export * from "./map";
-export * from "./sequence";
+ export { AttachState } from "@fluidframework/container-definitions";
```
Comment (Josmithr) [inline packages/framework/fluid-framework/src/index.ts:28]:
```suggestion
export { AttachState } from "@fluidframework/container-definitions";
```

**Comment by Josmithr** [comment]
> How will this affect packages that refer to the re-exported libraries directly? E.g. if we ever publish the container-definitions package, and other packages refer to AttachState directly in it vs through the fluid-framework package, will the links go to the right place?

The links will go to whichever package is referenced in the link. So if they use `{@link fluid-framework#AttachState}`, the link will go to the re-export, but if they use `{@link fluid-container#AttachState}`, the link will go to the base definition.

File: docs/content/docs/updates/pre-v1.0.0.md (hunk 2 above)
Comment (tylerbutler) [reply docs/content/docs/updates/pre-v1.0.0.md:244]: It's here: https://github.com/microsoft/FluidFramework/blob/main/docs/layouts/shortcodes/apiref.html

Shortcode docs here: https://gohugo.io/content-management/shortcodes/#readout

File: packages/framework/fluid-framework/package.json [hunk 6]
```diff
@@ -15,7 +15,7 @@
   "module": "lib/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "build": "concurrently npm:build:compile npm:lint",
+    "build": "concurrently npm:build:compile npm:lint && npm run build:docs",
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply packages/framework/fluid-framework/package.json:18]: @tylerbutler is this not the right place for this script to run? Other packages run this as a part of their standard build, don't they?

File: docs/content/docs/updates/pre-v1.0.0.md (hunk 2 above)
Comment (Josmithr) [reply docs/content/docs/updates/pre-v1.0.0.md:244]: Awesome! Thank you!

File: packages/framework/fluid-framework/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (Josmithr) [reply packages/framework/fluid-framework/package.json:18]: This seems to match the pattern in at least some of the other framework packages, and CI hasn't complained. So I think this is okay? Please feel free to re-open if not.

File: docs/_includes/links.md (hunk 1 above)
Comment (Josmithr) [reply docs/_includes/links.md:10]: For standard in-doc links, `apiref` should continue to work, but for these I think we will have to be more explicit.

File: packages/framework/fluid-framework/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/framework/fluid-framework/package.json:18]: fluid-build does some validation  of the build scripts and has carveouts for some packages. But in this case, it should already be doing the right thing:

https://github.com/microsoft/FluidFramework/blob/b97e90f4d5cd40eee10d5cb630b0bc6cb26b00ae/build-tools/packages/build-tools/src/fluidBuild/fluidPackageCheck.ts#L265-L277

**Review by tylerbutler — APPROVED** [review]

**Comment by Josmithr** [comment]
Update package README

File: docs/content/docs/updates/pre-v1.0.0.md [hunk 7]
```diff
@@ -238,32 +237,6 @@ better reflect their experimental state. If you use these classes, you can add a
...
-[AzureClient]: {{< relref "/docs/apis/azure-client/AzureClient-class.md" >}}
-[TinyliciousClient]: {{< relref "/docs/apis/tinylicious-client/TinyliciousClient-class.md" >}}
-
-[FluidContainer]: {{< relref "/docs/apis/fluid-static/fluidcontainer-class.md" >}}
-[IFluidContainer]: {{< relref "/docs/apis/fluid-static/ifluidcontainer-interface.md" >}}
-
-<!-- AUTO-GENERATED-CONTENT:END -->
+- Various APIs in the {{< packageref "azure-client" >}} and {{< packageref "tinylicious-client" >}} now return {{< apiref
+  "IFluidContainer" "interface" >}}s instead of {{< apiref "FluidContainer" "class" >}}s.
+  This change should have no effect to developers since the runtime object is the same ({{< issue 7457 >}}).
```
Comment (Josmithr) [inline docs/content/docs/updates/pre-v1.0.0.md:242]: Links should not have been removed.

**Review by ssimic2 — APPROVED** [review]

**Comment by Josmithr** [comment]
I am pulling out some of the clean-up aspects of this PR into separate PRs:
- #11925
- #11923

**Comment by Josmithr** [comment]
Pulling the package description update portion out into its own PR as well: #11975
